From 023732c7c17086cf5781a8a98b632495ffb48af4 Mon Sep 17 00:00:00 2001
From: Sam James <sam@gentoo.org>
Date: Thu, 17 Sep 2026 02:04:11 +0100
Subject: [PATCH 45/63] UPSTREAM: Revert "io: drop nonnull attribute for
 fchmodat, faccessat, fchownat's path argument [BZ #34313]"

This reverts commit b14c4b0bc95c25f9718807be07ad9a13442d2b6f.
---
 io/sys/stat.h  | 2 +-
 posix/unistd.h | 4 ++--
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/io/sys/stat.h b/io/sys/stat.h
index ff70b69f59..3069e187b0 100644
--- a/io/sys/stat.h
+++ b/io/sys/stat.h
@@ -370,7 +370,7 @@ extern int fchmod (int __fd, __mode_t __mode) __THROW;
    the directory FD is open on.  */
 extern int fchmodat (int __fd, const char *__file, __mode_t __mode,
 		     int __flag)
-     __THROW __wur;
+     __THROW __nonnull ((2)) __wur;
 #endif /* Use ATFILE.  */
 
 
diff --git a/posix/unistd.h b/posix/unistd.h
index dc5c2ec009..06a6a88b52 100644
--- a/posix/unistd.h
+++ b/posix/unistd.h
@@ -307,7 +307,7 @@ extern int execveat (int __fd, const char *__path, char *const __argv[],
    If AT_EACCESS is set in FLAG, then use effective IDs like `eaccess',
    otherwise use real IDs like `access'.  */
 extern int faccessat (int __fd, const char *__file, int __type, int __flag)
-     __THROW __wur;
+     __THROW __nonnull ((2)) __wur;
 #endif /* Use GNU.  */
 
 
@@ -510,7 +510,7 @@ extern int lchown (const char *__file, __uid_t __owner, __gid_t __group)
    on.  */
 extern int fchownat (int __fd, const char *__file, __uid_t __owner,
 		     __gid_t __group, int __flag)
-     __THROW __wur;
+     __THROW __nonnull ((2)) __wur;
 #endif /* Use GNU.  */
 
 /* Change the process's working directory to PATH.  */
-- 
2.55.0

